src/firstvoices/backend/models/dictionary.py

```diff
@@ -140,13 +142,39 @@ class TypeOfDictionaryEntry(models.TextChoices):
         related_name="incoming_related_dictionary_entries",
     )
 
+    # from fvcharacter:related_words
+    related_characters = models.ManyToManyField(
+        Character,
+        blank=True,
+        through="DictionaryEntryRelatedCharacter",
+        related_name="dictionary_entries",
+    )
+
     class Meta:
         verbose_name = _("Dictionary Entry")
         verbose_name_plural = _("Dictionary Entries")
 
     def __str__(self):
         return self.title
 
+    def save(self, *args, **kwargs):
+        mapper, created = AlphabetMapper.objects.get_or_create(site_id=self.site_id)
+
+        self.clean_title(mapper)
+        self.set_custom_order(mapper)
+        super().save(*args, **kwargs)
+
+    def clean_title(self, mapper):
+        # strip whitespace and normalize
```

Comment:
Would it make sense to move this function to the AlphabetMapper and make it return the cleaned string? It could be called something generic like normalize (or maybe.... ultranormalize). It needs the mapper anyway, and it seems like the kind of cleaning operation we may want to reuse for things like search query inputs or song titles.

Reply:
The `clean_confusables` part of this function is only necessary for sorting, and is on the AlphabetMapper already. We don't currently apply this to any non-sorted fields, so I would hesitate to extend it. But the NFC and stripping operations could definitely get pulled out to a more generic place so that we can later apply them to all user input or text fields.